include/util.h

```diff
@@ -1,10 +1,34 @@
 #pragma once
 
+typedef long long int ll;
```

Comment:
I'm not so sure we want these in the header file unless you want these typedefs accessible everywhere. Maybe an anonymous namespace?

Reply:
probably true, I'll add them to `cpp` files where I use them